app/controllers/items_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Merchant::ItemsController < ApplicationController
+  before_action :set_merchant
+  before_action :set_item, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @items = @merchant.items
+    @item_statuses = {}
+
+    @items.each do |item|
+      @item_statuses[item.id] = item.enabled? ? "Enabled" : "Disabled"
```

Comment:
woah! can you walk me through this and what it means / how it works?

Reply:
Sure!  We have a hash with item.id keys => status values. Then we're iterating over each item and checking whether enabled is true or false, and then assigning that status to the item.id key in the hash.  Then we can retrieve status of an item by calling its id from the hash.